src/java/org/apache/cassandra/schema/CompressionParams.java

```diff
@@ -514,12 +659,15 @@ public void serialize(CompressionParams parameters, DataOutputPlus out, int vers
                 out.writeUTF(entry.getValue());
             }
             out.writeInt(parameters.chunkLength());
-            out.writeInt(parameters.maxCompressedLength);
+            if (version >= MessagingService.VERSION_40)
+                out.writeInt(parameters.maxCompressedLength);
+            else
+                if (parameters.maxCompressedLength != Integer.MAX_VALUE)
+                    throw new UnsupportedOperationException("Cannot stream SSTables with uncompressed chunks to pre-4.0 nodes.");
```

Comment:
we don't stream anything to pre-4.0 nodes ever anyway.
go back to the `assert` approach IMHO.